shadowsocks/tcp.go

```diff
@@ -210,9 +213,12 @@ func (s *tcpService) Start() {
 			timeToCipher = time.Now().Sub(findStartTime)
 
 			if err != nil {
+				logger.Debugf("Failed to find a valid cipher after reading %v bytes: %v", proxyMetrics.ClientProxy, err)
+				io.Copy(ioutil.Discard, clientConn) // drain socket
```

Comment:
Add a comment like "Keep the connection open until we hit the authentication deadline to protect against probing attacks"

Reply:
done